tool/tsh/db.go

```diff
@@ -1076,13 +1081,41 @@ func isLocalProxyAlwaysRequired(protocol string) bool {
 	switch protocol {
 	case defaults.ProtocolSQLServer,
 		defaults.ProtocolSnowflake,
-		defaults.ProtocolCassandra:
+		defaults.ProtocolCassandra,
+		defaults.ProtocolDynamoDB:
 		return true
 	default:
 		return false
 	}
 }
 
+// formatDbCmdUnsupportedDBProtocol is a helper func that formats the unsupported DB protocol error message.
+func formatDbCmdUnsupportedDBProtocol(cf *CLIConf, database *tlsca.RouteToDatabase) string {
+	return fmt.Sprintf(dbCmdUnsupportedDBProtocol,
+		cf.CommandWithBinary(),
+		defaults.ReadableDatabaseProtocol(database.Protocol),
+		getDbCmdAlternatives(database))
+}
+
+// formatDbCmdUnsupportedTLSRouting is a helper func that formats the unsupported TLS routing error message.
+func formatDbCmdUnsupportedTLSRouting(cf *CLIConf, database *tlsca.RouteToDatabase) string {
+	return fmt.Sprintf(dbCmdUnsupportedTLSRouting,
+		cf.CommandWithBinary(),
+		defaults.ReadableDatabaseProtocol(database.Protocol),
+		getDbCmdAlternatives(database))
+}
+
+// getDbCmdAlternatives is a helper func that returns alternative tsh commands for connecting to a database.
+func getDbCmdAlternatives(database *tlsca.RouteToDatabase) string {
+	switch database.Protocol {
+	case defaults.ProtocolDynamoDB:
+		// DynamoDB only works with a local proxy tunnel and there is no "shell-like" cli, so `tsh db connect` doesn't make sense.
+		return `"tsh proxy db --tunnel"`
+	default:
+		return `"tsh db connect" or "tsh proxy db --tunnel"`
```

Comment:
Sounds good to me, I'll try refactoring to be similar to the login templating logic which prints connect/proxy/config commands conditionally using those `formatDatabase*` helpers

Reply:
```
[I] [14:01:59] gavin@mac ~/teleport-config [1] 
$ tsh db env mysql                                                                                                                     (base) 
ERROR: "tsh db env" is not supported for MySQL databases when TLS routing is enabled on the Teleport Proxy Service.

Please use one of the following commands to connect to the database:
    `tsh db connect --cluster=mars.local.gd --db-user=<user> mysql`
    `tsh proxy db --tunnel --cluster=mars.local.gd --db-user=<user> mysql`

[I] [14:02:02] gavin@mac ~/teleport-config [1] 
$ tsh db config cass
ERROR: "tsh db config" is not supported for Cassandra databases.

Please use one of the following commands to connect to the database:
    `tsh db connect --cluster=mars.local.gd --db-user=<user> cass`
    `tsh proxy db --tunnel --cluster=mars.local.gd --db-user=<user> cass`

[I] [14:02:06] gavin@mac ~/teleport-config [1] 
$ tsh db login --db-user=GavinDynamoDBRole dynamodb
Connection information for database "dynamodb" has been saved.

You can start a local proxy for database GUI clients:

  tsh proxy db --tunnel --cluster=mars.local.gd dynamodb

[I] [14:02:36] gavin@mac ~/teleport-config  
$ tsh db connect dynamodb
ERROR: "tsh db connect" is not supported for DynamoDB databases.

Please use the following command to connect to the database:
    `tsh proxy db --tunnel --cluster=mars.local.gd dynamodb`
[I] [14:02:40] gavin@mac ~/teleport-config [1] 
$ tsh proxy db dynamodb 
ERROR: "tsh proxy db" is not supported for DynamoDB databases without the --tunnel flag.

Please use the following command to connect to the database:
    `tsh proxy db --tunnel --cluster=mars.local.gd dynamodb`
```

How's this?

Reply:
I am not very good at UX either. Does `--cluster` flag always show up or only when `--cluster` is passed in? And personally I would also prefer without the back quotes so it's easier to copy/paste

Reply:
Oh, it shows the cluster because I had the env var `TELEPORT_CLUSTER` set. Without that it doesn't show the cluster flag.

I'll remove the backticks though

Reply:
`--db-user=` flag only shows up because I didn't specify it when I did `tsh db login`, else that will also not appear in the suggested command.

Reply:
Sounds good!